src/main/kotlin/org/jetbrains/research/testspark/appstarter/TestSparkStarter.kt

```diff
@@ -0,0 +1,219 @@
+                    val targetPsiClass = detectPsiClass(psiFile.classes, classUnderTestName) ?: run {
+                        println("couldn't find $classUnderTestName in $filePath")
+                        exitProcess(1)
+                    }
+
+                    println("PsiClass ${targetPsiClass.qualifiedName} is detected! Start the test generation process.")
+
+                    // update settings
+                    val settingsState = project.getService(LLMSettingsService::class.java).state
+                    settingsState.currentLLMPlatformName = LLMDefaultsBundle.get("grazieName")
+                    settingsState.grazieToken = token
+                    settingsState.grazieModel = model
+                    settingsState.classPrompts = JsonEncoding.encode(mutableListOf(File(promptTemplateFile).readText()))
+                    settingsState.junitVersion = when (jUnitVersion.filter { it.isDigit() }) {
+                        "4" -> JUnitVersion.JUnit4
+                        "5" -> JUnitVersion.JUnit5
+                        else -> {
+                            throw IllegalArgumentException("JUnit version $jUnitVersion is not supported. Supported JUnit versions are '4' and '5'")
+                        }
+                    }
+                    project.service<PluginSettingsService>().state.buildPath = classPath
+
+                    // Prepare Project Context
+                    // First, get CUT Module
+                    val cutModule =
+                        ProjectFileIndex.getInstance(project)
+                            .getModuleForFile(targetPsiClass.containingFile.virtualFile)!!
+                    // Then, instantiate the project context
+                    val projectContext = ProjectContext(
+                        classPath,
+                        output,
+                        targetPsiClass,
+                        targetPsiClass.qualifiedName,
+                        cutModule
+                    )
+                    // Prepare the test generation data
+                    val testGenerationData = TestGenerationData(
+                        resultPath = output,
+                        testResultName = "HeadlessGeneratedTests"
+                    )
+                    println("[TestSpark Starter] Indexing is done")
+
+                    // get package name
+                    val packageList = targetPsiClass.qualifiedName.toString().split(".").toMutableList()
+                    packageList.removeLast()
+                    val packageName = packageList.joinToString(".")
+
+                    // Create a process Manager
+                    val llmProcessManager = Llm()
+                        .getLLMProcessManager(
+                            project,
+                            psiFile,
+                            targetPsiClass.textRange.startOffset,
+                            ""
```

Comment:
What does this last `""` correspond to? It is better to provide it as a named function argument.